### PR TITLE
feat: sanitize form-post html attr

### PR DESCRIPTION
**Pull request by 0xPenryn**
Sanitizes form-post HTML attributes to alphanumeric characters, `.`, `_`, and `-`.

Completes DEV-1040

### CODE REVIEW

File: src/app/authenticate/route.ts [hunk 1]
```diff
@@ -64,7 +64,10 @@ export const POST = async (req: NextRequest): Promise<NextResponse> => {
         <form id="formRedirect" method="post" action="${url}">    
           ${Array.from(result.url_params.entries()).map(
             ([key, value]) =>
-              `<input type="hidden" name="${key}" value="${value}" />`
+              `<input type="hidden" name="${key.replace(
```
Comment (andy-t-wang) [inline src/app/authenticate/route.ts:67]: can the user even set the keys?

**Review by andy-t-wang — APPROVED** [review]

**Comment by ph3t** [comment]
@0xPenryn Won't this break some of the parameters like redirect_uri? Instead, I'd vouch for output escaping by replacing special characters with their respective HTML entity:
```
        .replace(/&/g, '&amp;')
        .replace(/</g, '&lt;')
        .replace(/>/g, '&gt;')
        .replace(/"/g, '&quot;')
        .replace(/'/g, '&#39;')
        .replace(/\//g, '&#x2F;');
```

**Comment by 0xPenryn** [comment]
> @0xPenryn Won't this break some of the parameters like redirect_uri? Instead, I'd vouch for output escaping by replacing special characters with their respective HTML entity:
> 
> ```
>         .replace(/&/g, '&amp;')
>         .replace(/</g, '&lt;')
>         .replace(/>/g, '&gt;')
>         .replace(/"/g, '&quot;')
>         .replace(/'/g, '&#39;')
>         .replace(/\//g, '&#x2F;');
> ```

@ph3t the only parameters possibly returned in the result url_params are `code`, `token`, `id_token`, and `state`.

`code` is a hex string, `token` and `id_token` are JWT's, and `state` is already validated and limited to the same character set. The restrictions as set should not impact existing functionality.

**Comment by andy-t-wang** [comment]
Yeah I think the redirect URL is inside of the action field in the form

**Comment by ph3t** [comment]
> @ph3t the only parameters possibly returned in the result url_params are `code`, `token`, `id_token`, and `state`.

I'm aware, but I'd like to prevent issues like [DEV-1005](https://linear.app/worldcoin/issue/DEV-1005/insufficient-filtering-of-state-parameter-in-response-mode-form-post) if someone adds other unvalidated parameters to the response in the future.

**Review by ph3t — APPROVED** [review]

**Pull request merged**